lib/Net/Google/Drive/Simple.pm

```diff
@@ -262,6 +262,45 @@ sub file_upload {
     return $file_id;
 }
 
+###########################################
+sub rename {
+###########################################
+    my ( $self, $file_id, $new_name ) = @_;
+
+    my $url = URI->new( $self->{api_file_url} . "/$file_id" );
+
+    my $req = &HTTP::Request::Common::PUT(
```

Comment:
would be a bonus to have one `http_put` helper and use it, similar to `http_delete`
you would pass one hashref to it which would be everything after  authorization_headers
`$self->http_put( $url, { Accept => "application/json", ... } )`

Reply:
Happy to take a look at this. The function would be called by both `sub rename` and `sub file_upload`, so calls by the latter would involve passing the contents of the file to the helper sub.

I.e. we could easily have 100MB or 1GB of data in the hashref - are there any performance considerations that need to be considered? I have no idea - that's why I'm asking.

Reply:
as you said it s a `dataref` which means it's a `ref`/`reference` to the hash... there is only one single hash. The reference is just a pointer to the data structure, there is no harms in passing pointers. Usually we want to limit the number of arguments as in most language for performance purpose, and avoiding spamming the stack.

Reply:
Please see my commit 5728e16aa8e9413bed5146d37ec5d49a7697db60